common/src/autogluon/common/utils/utils.py

```diff
@@ -5,21 +5,40 @@
 import os
 import platform
 import sys
+import numpy as np
 from typing import Dict, Any
 
 logger = logging.getLogger(__name__)
 
 
+def setup_result_path(checkpoint_name):
+    utcnow = datetime.utcnow()
+    timestamp = utcnow.strftime("%Y%m%d_%H%M%S")
+    result_path = f"Result{os.path.sep}ag-{timestamp}{os.path.sep}" f"{checkpoint_name}{os.path.sep}"
+    for i in range(1, 1000):
```

Comment:
I'm curious to know the scenario here, why it would create so many jobs/folders in one second?

Reply:
Yea I saw this in one of the previously implemented util functions `setup_outputdir`. I speculate that this was to run retry creating folders on failure. Should we delete something like this?

Reply:
But it looks like in 1000 attempts the result_path stays the same? It's not updated with `-{i:03d}`

Reply:
Also this may cause a mismatch between model save path and model result path, since they have different timestamp and `-{i:03d}` postfix. Shall we store the result in result folder of the predictor's `self._save_path`? @bryanyzhu

Reply:
I think it is best to save the result to the same folder that checkpoint/logs are saved

Reply:
Yea I agree. What is the self._save_path? Is this the folder where the checkpoint is saved? @FANGAreNotGnu

Reply:
Yes

Reply:
I just checked the init in `MultiModalPredictor`. It does seem to be None.